Intergrate unit assets

esbudylin opened the pull request:
This PR
- Removes the C7 assets from the repository and adds the [assets repository](https://github.com/C7-Game/Assets) as a submodule.
- Introduces a new json save for the standalone mode. It excludes the units we don't have graphics for and updates art names for units that got an art replacement.
- Changes the BarbarianInfo class: previously, when loading data on barbarian units from json, this class relied on a unit's position in the list of all units. This PR makes it rely on a unit name instead, which makes it easier to make changes to the unit list.

There are a few issues with the standalone mode:
- Some unit sprites are displayed with wrong offsets. I've noticed this issue with warrior, horseman, and curragh, but I haven't tested all the units in game.  
- There are no tech box textures for different sizes and eras.

A screenshot of the game running standalone:
<img width="1920" height="1032" alt="image_2025-10-22_20-50-18" src="https://github.com/user-attachments/assets/05236ecb-9d44-4857-93ca-ca6621b0b72d" />

esbudylin committed:
Add a json save for standalone mode

- Create a JSON save for standalone mode: exclude the units we don't
have graphics for, and update art names for units that will get a
replacement.

- Update BarbarianInfo class, so it no longer relies on unit indexes and
uses unit names instead.

esbudylin committed:
Update search path for C7 assets

esbudylin committed:
Remove C7 art, add Assets as submodule

TomWerner reviewed (approved):
Nice, just a couple questions:
 - Do we need to update the dev instructions? I had to run `git submodule init` and `git submodule update` to get the assets loaded
 - Should we update `SaveTest` to run with the old and new save files?

esbudylin committed:
Merge branch 'Development' of github.com:C7-Game/Prototype into yegor/integrate-unit-assets

esbudylin committed:
Merge branch 'Development' of github.com:C7-Game/Prototype into yegor/integrate-unit-assets

esbudylin committed:
Run SaveTest against standalone mode save file

esbudylin committed:
Merge branch 'Development' of github.com:C7-Game/Prototype into yegor/integrate-unit-assets

esbudylin committed:
Update CI/CD job to work with the Assets submodule

esbudylin committed:
Fix case-insensitive file lookup

Previous version incorrectly used "**" as a wildcard instead of "*".

See C# docs: https://learn.microsoft.com/en-us/dotnet/api/system.io.directory.enumeratefilesystementries?view=net-9.0

esbudylin committed:
Fix loading unit assets in exported debug builds

Previously, loading unit assets in debug builds didn't work, due to
the fact that ProjectSettings.GlobalizePath() doesn't work in an
exported projects

See Godot docs: https://docs.godotengine.org/en/stable/classes/class_projectsettings.html#class-projectsettings-method-globalize-path

esbudylin commented:
I've created a wiki page documenting the difference between standard and standalone modes and how to set up a submodule: https://github.com/C7-Game/Prototype/wiki/Standard-and-standlone-modes I'll go ahead and merge this. I think if the submodule becomes unhandy, we can return to this discussion.
